src/Angor/Shared/Services/NetworkMonitoringService.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Angor.Shared.Services;
+
+public class NetworkMonitoringService : IDisposable
+{
+    private readonly IServiceProvider _serviceProvider;
+
+    public NetworkMonitoringService(IServiceProvider serviceProvider)
+    {
+        _serviceProvider = serviceProvider;
+    }
+
+    public async Task CheckAndEnsureServices()
+    {
+        using (var scope = _serviceProvider.CreateScope())
```

Comment:
network service is scoped? how is the scoped lifetime manages in a wasm app I wonder?

Reply:
```
Scoped: The service is created once for the duration of the user’s session in the browser. Essentially, a scoped service in Blazor WebAssembly acts as a singleton within the context of a single user's session.
This means that once a scoped service is created, it remains the same for the entirety of the user's interaction with the Blazor WebAssembly app.
```

There is no real point in using scoped in wasm @DavidGershony